### PR TITLE
carma-platform is not using dynamic dependency in its CI

### PR DESCRIPTION
**Pull request by MishkaMN**
# PR Details
## Description

Related to this is the CI.yml where it is not using the dynamic branch selection we introduced and is hardcoded to run on autoware.ai:develop as well as its checkout script is using hardcoded BRANCH=develop line in checkout.sh script that is meant for failsafe.
Please see the original Ticket for more details:
[CAR-6034](https://usdot-carma.atlassian.net/browse/CAR-6034)

## Related GitHub Issue
NA

## Related Jira Key
[CAR-6034](https://usdot-carma.atlassian.net/browse/CAR-6034)

## Motivation and Context
CI is using hardcoded branches and images after master release of 4.5.0

## How Has This Been Tested?
tested on the PR itself and 
https://github.com/usdot-fhwa-stol/carma-platform/pull/2382 OR
https://github.com/usdot-fhwa-stol/carma-platform/actions/runs/9021704219
where I tested against release/lavida (which shows failed due to some branches not existing anymore, but the correct branch is selected in the CI)

## Types of changes

- [X] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [X] I have added any new packages to the sonar-scanner.properties file
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

[CAR-6034]: https://usdot-carma.atlassian.net/browse/CAR-6034?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CAR-6034]: https://usdot-carma.atlassian.net/browse/CAR-6034?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/Ci.yml [hunk 1]
```diff
@@ -6,7 +6,19 @@ on:
   push:
     branches: [develop, master]
 jobs:
+  # First job: Preliminary setup or determinations
+  preliminary:
```
Comment (paulbourelly999) [inline .github/workflows/Ci.yml:10]: Can we update this step to be named the same way it is named in other branches

Comment (paulbourelly999) [reply .github/workflows/Ci.yml:10]: https://github.com/usdot-fhwa-stol/carma-messenger/blob/76b2b7436d46ef64b67ff2fa362f2c802d51fa6c/.github/workflows/ci.yml#L11

**Review by paulbourelly999 — CHANGES_REQUESTED** [review]
Small change requested

File: .github/workflows/Ci.yml (hunk 1 above)
Comment (MishkaMN) [reply .github/workflows/Ci.yml]: remamed

**Pull request merged**